src/sagemaker_pytorch_serving_container/torchserve.py

```diff
@@ -0,0 +1,215 @@
+MODEL_STORE = "/" if ENABLE_MULTI_MODEL else DEFAULT_TS_MODEL_DIRECTORY
+
+PYTHON_PATH_ENV = "PYTHONPATH"
+REQUIREMENTS_PATH = os.path.join(code_dir, "requirements.txt")
+TS_NAMESPACE = "org.pytorch.serve.ModelServer"
+
+
+def start_torchserve(handler_service=DEFAULT_HANDLER_SERVICE):
+    """Configure and start the model server.
+
+    Args:
+        handler_service (str): python path pointing to a module that defines
+            a class with the following:
+
+                - A ``handle`` method, which is invoked for all incoming inference
+                    requests to the model server.
+                - A ``initialize`` method, which is invoked at model server start up
+                    for loading the model.
+
+            Defaults to ``sagemaker_pytorch_serving_container.default_handler_service``.
+
+    """
+
+    if ENABLE_MULTI_MODEL:
+        if not os.getenv("SAGEMAKER_HANDLER"):
+            os.environ["SAGEMAKER_HANDLER"] = handler_service
+        _set_python_path()
+    else:
+        _adapt_to_ts_format(handler_service)
+
+    _create_torchserve_config_file()
+
+    if os.path.exists(REQUIREMENTS_PATH):
+        _install_requirements()
+
+    ts_torchserve_cmd = [
+        "torchserve",
+        "--start",
+        "--model-store",
+        MODEL_STORE,
+        "--ts-config",
+        TS_CONFIG_FILE,
+        "--log-config",
+        DEFAULT_TS_LOG_FILE,
+        "--models",
+        "model.mar"
+    ]
+
+    print(ts_torchserve_cmd)
+
+    logger.info(ts_torchserve_cmd)
+    subprocess.Popen(ts_torchserve_cmd)
+
+    ts_process = _retrieve_ts_server_process()
+
+    _add_sigterm_handler(ts_process)
+
+    ts_process.wait()
+
+
+def _adapt_to_ts_format(handler_service):
+    if not os.path.exists(DEFAULT_TS_MODEL_DIRECTORY):
+        os.makedirs(DEFAULT_TS_MODEL_DIRECTORY)
+
+    model_archiver_cmd = [
+        "torch-model-archiver",
+        "--model-name",
+        DEFAULT_TS_MODEL_NAME,
+        "--handler",
+        handler_service,
+        "--serialized-file",
+        os.path.join(environment.model_dir, DEFAULT_TS_MODEL_SERIALIZED_FILE),
+        "--export-path",
+        DEFAULT_TS_MODEL_DIRECTORY,
+        "--extra-files",
+        os.path.join(environment.model_dir, environment.Environment().module_name + ".py"),
+        "--version",
+        "1",
+    ]
+
+    logger.info(model_archiver_cmd)
+    subprocess.check_call(model_archiver_cmd)
+
+    _set_python_path()
+
+
+def _set_python_path():
+    # Torchserve handles code execution by appending the export path, provided
+    # to the model archiver, to the PYTHONPATH env var.
+    # The code_dir has to be added to the PYTHONPATH otherwise the
+    # user provided module can not be imported properly.
+    code_dir_path = "{}:".format(environment.code_dir)
+
+    if PYTHON_PATH_ENV in os.environ:
+        os.environ[PYTHON_PATH_ENV] = code_dir_path + os.environ[PYTHON_PATH_ENV]
+    else:
+        os.environ[PYTHON_PATH_ENV] = code_dir_path
```

Comment:
using format strings or `join()` is more efficient than string concatenation

```suggestion
    if PYTHON_PATH_ENV in os.environ:
        os.environ[PYTHON_PATH_ENV] = "{}:{}".format(environment.code_dir, os.environ[PYTHON_PATH_ENV])
    else:
        os.environ[PYTHON_PATH_ENV] = environment.code_dir
```